google/cloud/spanner_dbapi/connection.py

```diff
@@ -187,15 +229,16 @@ def _rerun_previous_statements(self):
                     self.connection._transaction = None
                     raise Aborted(status.details)
 
-                retried_checksum = ResultsChecksum()
-                retried_checksum.consume_result(res)
-                retried_checksum.consume_result(status.code)
+                if not self.read_only:
+                    retried_checksum = ResultsChecksum()
+                    retried_checksum.consume_result(res)
+                    retried_checksum.consume_result(status.code)
 
-                _compare_checksums(checksum, retried_checksum)
+                    _compare_checksums(checksum, retried_checksum)
             else:
                 res_iter, retried_checksum = self.run_statement(statement, retried=True)
                 # executing all the completed statements
-                if statement != self._statements[-1]:
+                if statement != self._statements[-1] and not self.read_only:
```

Comment:
Here also: This should not be necessary. We should never call _rerun_previous_statements unless the transaction is a multi-use read/write transaction.